### PR TITLE
Fix bucket lock not being released on errors while being acquired

### PR DESCRIPTION
**Pull request by davfsa**
### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: tests/hikari/impl/test_buckets.py [hunk 1]
```diff
@@ -44,20 +44,27 @@ def compiled_route(self, template):
 
     @pytest.mark.asyncio()
     async def test_async_context_manager(self, compiled_route):
-        with mock.patch.object(asyncio, "Lock") as lock:
-            lock.return_value.acquire = mock.AsyncMock()
-            with mock.patch.object(buckets.RESTBucket, "acquire", new=mock.AsyncMock()) as acquire:
+        with mock.patch.object(buckets.RESTBucket, "acquire", new=mock.AsyncMock()) as acquire:
+            with mock.patch.object(buckets.RESTBucket, "release") as release:
```
Comment (FasterSpeeding) [inline tests/hikari/impl/test_buckets.py:48]: Same here with the exit stack

File: tests/hikari/impl/test_buckets.py [hunk 2]
```diff
@@ -94,9 +101,12 @@ async def test_acquire_when_unknown_bucket(self, compiled_route):
     async def test_acquire_when_too_long_ratelimit(self, compiled_route):
         with buckets.RESTBucket("spaghetti", compiled_route, 60) as rl:
             rl.reset_at = time.perf_counter() + 999999999999999999999999999
-            with mock.patch.object(buckets.RESTBucket, "is_rate_limited", return_value=True):
-                with pytest.raises(errors.RateLimitTooLongError):
-                    await rl.acquire()
+            with mock.patch.object(buckets.RESTBucket, "release") as release:
+                with mock.patch.object(buckets.RESTBucket, "is_rate_limited", return_value=True):
+                    with pytest.raises(errors.RateLimitTooLongError):
```
Comment (FasterSpeeding) [inline tests/hikari/impl/test_buckets.py:106]: contextlib.ExitStack pls

**Comment by FasterSpeeding** [comment]
Has this been tested in the real world?

File: hikari/impl/rest.py [hunk 3]
```diff
@@ -2115,9 +2122,9 @@ async def revoke_access_token(
         token: typing.Union[str, applications.PartialOAuth2Token],
     ) -> None:
         route = routes.POST_TOKEN_REVOKE.compile()
-        form = data_binding.URLEncodedFormBuilder()
-        form.add_field("token", str(token))
-        await self._request(route, form_builder=form, auth=self._gen_oauth2_token(client, client_secret))
+        form_builder = data_binding.URLEncodedFormBuilder()
+        form_builder.add_field("token", str(token))
```
[the remark this answers is not in the document]
Comment (davfsa) [reply hikari/impl/rest.py:2126]: I am going to wait for #971 before working on this one again

**Comment by davfsa** [comment]
> Has this been tested in the real world?

I did back when I opened it. A lot has changed since then so will probably have to revisit it

**Review by FasterSpeeding — APPROVED** [review]

**Pull request merged**